Form/ChoiceList/ModelChoiceLoader.php

```diff
@@ -77,10 +77,10 @@ public function __construct(ModelManagerInterface $modelManager, $class, $proper
     public function loadChoiceList($value = null)
     {
         if (!$this->choiceList) {
-            if (is_array($this->choices)) {
-                $entities = $this->choices;
-            } elseif ($this->query) {
+            if ($this->query) {
                 $entities = $this->modelManager->executeQuery($this->query);
+            } elseif (is_array($this->choices)) {
```

Comment:
For sonata_type_model it will be always true because of:
'choices'           => array(), in default options
which results in empty choice list in select box if you create form field:
$formMapper->add('groups', 'sonata_type_model', ['multiple' => true, 'required' => true])